pkg/xds/envoy/virtualhosts/v3/configurer.go

```diff
@@ -0,0 +1,36 @@
+package v3
+
+import (
+	envoy_route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+)
+
+// VirtualHostConfigurer is responsible for configuring a single aspect of the entire Envoy VirtualHost,
+// such as Route, CORS, etc.
+type VirtualHostConfigurer interface {
+	// Configure configures a single aspect on a given Envoy VirtualHost.
+	Configure(virtualHost *envoy_route.VirtualHost) error
+}
+
+// VirtualHostConfigureFunc adapts a configuration function to the
+// VirtualHostConfigurer interface.
+type VirtualHostConfigureFunc func(vh *envoy_route.VirtualHost) error
```

Comment:
I think we don't really need a `v3` subdirectory here, let's keep it at `pkg/xds/envoy/virtualhosts`

Reply:
I did that to keep the same pattern as it is done for the  route or listener. How is it different with VirtualHost ?

Reply:
Those are outdated patterns from when there was a transition from v2 to v3. There's unlikely to be a v4.